scanner_test.go

```diff
@@ -353,6 +353,64 @@ func TestScan(t *testing.T) {
 			),
 		},
 
+		{
+			name: "oct null",
+			text: "\"\\0\"",
```

Comment:
You may want to write this as `` `"\0"` `` instead. Go's raw strings use backticks.

Reply:
Oh FFS. Apparently I forgot that backticks were a thing. I originally tried to write these input strings using apostrophes instead and of course taht didn't work at all, which is why I went with the double quotes instead. Fixed here and also everywhere else now.